action/action-simulator/src/main/java/com/powsybl/action/simulator/parallel/ParallelLoadFlowActionSimulator.java

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) 2018, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.action.simulator.parallel;
+
+import com.powsybl.action.dsl.ActionDb;
+import com.powsybl.action.simulator.loadflow.LoadFlowActionSimulator;
+import com.powsybl.action.simulator.loadflow.LoadFlowActionSimulatorConfig;
+import com.powsybl.action.simulator.loadflow.LoadFlowActionSimulatorObserver;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.computation.*;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.security.SecurityAnalysisResult;
+import com.powsybl.security.SecurityAnalysisResultMerger;
+import com.powsybl.security.converter.SecurityAnalysisResultExporters;
+import com.powsybl.security.json.SecurityAnalysisResultDeserializer;
+import com.powsybl.tools.ToolRunningContext;
+import org.apache.commons.cli.CommandLine;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+
+import static com.powsybl.action.simulator.tools.ActionSimulatorToolConstants.*;
+
+public class ParallelLoadFlowActionSimulator extends LoadFlowActionSimulator {
+
+    private static final String ITOOLS_PRG = "itools";
+
+    private final int para;
```

Comment:
What "para" means? Could you find a better name?

Reply:
The number of sub-processes expected to launch simulator in parallel mode.
realPara is the used when total contingencies is smaller than the expected parallel counts,